### PR TITLE
Regex & Spec Fixes

### PR DESCRIPTION
**Pull request by ccmeyers**
Avoid pulling out capitalized namespaces separated by periods during extraction.

TODO:
- [ ] Publish new version on merge

### CODE REVIEW

File: test/example/missing-namespace.output.coffee [hunk 1]
```diff
@@ -1,4 +1,5 @@
 
+
```
Comment (juanca) [inline test/example/missing-namespace.output.coffee]: Double new line required?

File: index.js [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 module.exports = function loader(source) {
+  this.cacheable();
   const namespacesToReplace = {};
   const shortNamespaceToRequire = {};
   const namespaceToShort = {};
   const shortToNamespace = {};
+  const namespaceRegexp = namespace =>
+    // for now don't include a capital letter after a period
+    // because we haven't dealt with externals (i.e. `Notice.Alert`)
+     `([^\\w\\.])(${namespace})(\\s|\\.[a-z]|\\(|,)`;
```
Comment (juanca) [inline index.js]: Indentation is off... Kinda weird that the ES linter is not catching this?

File: package.json [hunk 3]
```diff
@@ -37,5 +45,8 @@
     "eslint-plugin-jsx-a11y": "^2.2.3",
     "eslint-plugin-react": "^6.7.1",
     "jasmine": "^2.5.2"
+  },
+  "bin": {
```
Comment (juanca) [inline package.json]: This is dope! Didn't know something like this existed for NPM package files. :+1:

File: test/index-spec.js [hunk 4]
```diff
@@ -68,7 +69,9 @@ describe('kitno-loader', () => {
   });
 });
 
+/* eslint-disable max-len */
 // Example of bad internal namespaces (because already exported?)
```
Comment (juanca) [inline test/index-spec.js]: This is internal to bigmaven.

Let's remove?

File: index.js [hunk 5]
```diff
@@ -98,10 +103,10 @@ module.exports = function loader(source) {
     if (shortToNamespace[actualClass]) {
       actualClass = namespaceDedupe(actualClass, defNames.join('.'));
     }
-    replacedSource = newSource.replace(definitionName, actualClass);
+    replacedSource = newSource.replace(new RegExp(definitionName, 'g'), actualClass);
     newSource = `${replacedSource}module.exports = ${actualClass}\n`;
   } else {
-    console.log('DAWG!', matches, definitionName)
+    console.log('DAWG!', matches, definitionName); // eslint-disable-line no-console
     // console.log('===========================');
```
Comment (juanca) [inline index.js]: If this loader is getting close to production-beta ready, should these unused console logs be cleaned up?

File: test/example/missing-namespace.output.coffee (hunk 1 above)
Comment (naganowl) [inline test/example/missing-namespace.output.coffee]: Unfortunately because of how/where it puts newlines, yes.

**Pull request merged**